src/DataManagement/Saving/AlternativeSavingCalculator.cs

```diff
@@ -30,9 +28,16 @@ public CalculationResults GetSuggestedExpensesOffers(IEntryList entryList, IGoal
             var averagesOfAmountByImportanceDefault = new decimal[enumCount];
             var entryAmounts = new int[enumCount];
 
+            if (entryList.Count == 0)
+            {
+                generateRandomData = true;              
+            }
+
             for (var i = enumCount; i > (int)Importance.Necessary; i--)
             {
-                var expenses = entryList.GetBy((Importance)i);
+                IEntryList expenses;
+                expenses = !generateRandomData ? entryList.GetBy((Importance)i) : expensesRandomList.GetBy((Importance)i);
+
                 var ratio = enumCount - i;
                 foreach (var entry in expenses)
                 {
```

Comment:
Can't comment on line 44 itself, but a way to simplify this would be:

       decimal amountAfterSaving;

(In this situation you don't need to assign a value with the declaration because you are overriding this value anyways in the code below it)

Reply:
hmm, but then we make the "var" into a "decimal", even if it does get rid of an unnecessary assignment, so why, exactly, is this better? The var is useless/unnecessary here?

Reply:
Well if I understand correctly, you want to avoid unnecessary value assignment as happens in this case ( basically if you check your code below, you could assign  
      
      var amountAfterSaving = 999999999M

and it would change how your code would work, so that's why I believe var should not be used in this case, even though we usually declare everything with var ( and to finish off Resharper is suggesting this change as well :D )

Reply:
you should use var when you're assigning something to newly created variable, since in this case you don't have what to assign, yet, it's better to use `decimal _name_;`

Reply:
oooh, ok, makes sense, i assumed incorrectly in a couple of ways :D